### PR TITLE
AGPUSH-1060 : Check for infinite redirect loops

### PR DESCRIPTION
**Pull request by sebastienblanc**
Inspired by https://github.com/smiklosovic/arquillian-spacelift/blob/c3f7a849eff6f78e8b5aaacfae0e9068b778e354/spacelift-impl/src/main/java/org/arquillian/spacelift/tool/basic/DownloadTool.java#L108-L172

@matzew @TadeasKriz can you review ?

### CODE REVIEW

**Comment by danielpassos** [comment]
@sebastienblanc wdyt to add a test for it?

**Comment by sebastienblanc** [comment]
@danielpassos Good idea, test has been added

File: src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java [hunk 1]
```diff
@@ -240,7 +242,14 @@ public void send(UnifiedMessage unifiedMessage) {
      * @param callback the {@link MessageResponseCallback} that will be called once the POST request completes.
      */
     private void submitPayload(String url, String jsonPayloadObject, String pushApplicationId, String masterSecret,
-            MessageResponseCallback callback) {
+            MessageResponseCallback callback, List<String> redirectUrls) {
+        if (redirectUrls.contains(url)) {
+            throw new IllegalStateException("The site contains an infinite redirect loop! Duplicate url: " +
+                    url);
+        } else {
+            redirectUrls.add(url);
```
Comment (matzew) [inline src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java]: hrm - what if there is no redirect at all ? Do we need to add the URL that early  ?

File: src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java [hunk 2]
```diff
@@ -240,7 +242,14 @@ public void send(UnifiedMessage unifiedMessage) {
      * @param callback the {@link MessageResponseCallback} that will be called once the POST request completes.
```
Comment (matzew) [inline src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java]: I'd perhaps update the JavaDoc as well

File: src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java (hunk 1 above)
Comment (sebastienblanc) [inline src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java]: yes but if there is an redirect then we save an "useless" post by doing this early. Not sure what is the best.

Comment (matzew) [inline src/main/java/org/jboss/aerogear/unifiedpush/DefaultPushSender.java]: yeah, @sebastienblanc let's keep it :-)

**Comment by abstractj** [comment]
@sebastienblanc mind to rebase? Two commits with "Update javadoc" makes me dizzy

**Comment by sebastienblanc** [comment]
@abstractj squashed the2 last commits

**Comment by matzew** [comment]
:+1: ship it !

**Pull request merged**